ruby/anagram.rb

```diff
@@ -0,0 +1,16 @@
+class Anagram
+  attr_reader :root_word
+
+  def initialize(root_word)
+    @root_word = root_word
+  end
+
+  def match(candidates)
+    anagrams = []
+    for c in candidates do
```

Comment:
This is a side-effect of me living in PythonLand for awhile. After refreshing myself on the `each`, I chose to use the `Array#select` after all to avoid assigning the empty array to a variable.